src/gui/embed.cpp

```diff
@@ -24,22 +24,78 @@
 
 #include "embed.h"
 
-#include <QDebug>
 #include <QDir>
+#include <QGuiApplication>
 #include <QImageReader>
+#include <QPainter>
 #include <QPixmapCache>
 #include <QResource>
+#include <QScreen>
+#include <QSvgRenderer>
 
 namespace lmms::embed {
 
 namespace {
 
+// QPixmapCache and HiDPI compatible SVG-->QPixmap wrapper
+auto loadSvgPixmap(const QString& resourceName, int width, int height) -> QPixmap
+{
+	// QFile requires the file extension to be present, unlike QImageReader
+	QString fileName = resourceName;
+	if (!fileName.endsWith(".svg", Qt::CaseInsensitive)) {
+		fileName += ".svg";
+	}
```

Comment:
> 4. Block braces SHOULD be on their own line

Source: [LMMS Coding Conventions](https://github.com/LMMS/lmms/wiki/Coding-Conventions#whitespace)

Alternatively, this is short enough it could be on one line (which I personally prefer).

```suggestion
	if (!fileName.endsWith(".svg", Qt::CaseInsensitive)) { fileName += ".svg"; }
```